pkg/services/ngalert/eval/eval.go

```diff
@@ -157,6 +157,24 @@ func (evalResults Results) HasErrors() bool {
 	return false
 }
 
+// HasNonRetryableErrors returns true if we have at least 1 result with:
+// 1. A `State` of `Error`
+// 2. The `Error` attribute is not nil
+// 3. The `Error` type is of `&invalidEvalResultFormatError`
+// Our thinking with this approach, is that we don't want to retry errors that have relation with invalid alert definition format.
+func (evalResults Results) HasNonRetryableErrors() bool {
+	var count int
+	for _, r := range evalResults {
+		if r.State == Error && r.Error != nil {
+			if errors.Is(r.Error, &invalidEvalResultFormatError{}) {
+				count++
```

Comment:
Nit: You can return true here and finish the loop early. You then also don't need the counter.

Reply:
Thanks!